solr/core/src/java/org/apache/solr/search/SolrReturnFields.java

```diff
@@ -532,10 +540,18 @@ private void addField(
     okFieldNames.add(key);
     // a valid field name
     if (SCORE.equals(field)) {
-      _wantsScore = true;
+      if (_noRows) {
```

Comment:
At least add a TODO to not even get to this point if noRows is true; we shouldn't even parse "fl".  As a reader of this code, it's odd to see special handling on SCORE in particular just because of noRows.